scripts/sct_dmri_moco.py

```diff
@@ -40,78 +41,107 @@ def get_parser():
-    parser.add_option(name='-param',
-                      type_value=[[','], 'str'],
-                      description="Advanced parameters. Assign value with \"=\"; Separate arguments with \",\"\n"
-                                  "poly [int]: Degree of polynomial function used for regularization along Z. For no regularization set to 0. Default=" + param_default.poly + ".\n"
-                                                "smooth [mm]: Smoothing kernel. Default=" + param_default.smooth + ".\n"
-                                                  "metric {MI, MeanSquares, CC}: Metric used for registration. Default=" + param_default.metric + ".\n"
-                                                  "gradStep [float]: Searching step used by registration algorithm. The higher the more deformation allowed. Default=" + param_default.gradStep + ".\n"
-                                                    "sample [None or 0-1]: Sampling rate used for registration metric. Default=" + param_default.sampling + ".\n",
-                      mandatory=False)
-    parser.add_option(name='-x',
-                      type_value='multiple_choice',
-                      description='Final Interpolation.',
-                      mandatory=False,
-                      default_value=param_default.interp,
-                      example=['nn', 'linear', 'spline'])
-    parser.add_option(name='-ofolder',
-                      type_value='folder_creation',
-                      description='Output folder',
-                      mandatory=False,
-                      default_value=param_default.path_out,
-                      example='dmri_moco_results/')
-    parser.usage.addSection('MISC')
-    parser.add_option(name="-r",
-                      type_value="multiple_choice",
-                      description='Remove temporary files.',
-                      mandatory=False,
-                      default_value='1',
-                      example=['0', '1'])
-    parser.add_option(name="-v",
-                      type_value='multiple_choice',
-                      description="verbose: 0 = nothing, 1 = classic, 2 = expended",
-                      mandatory=False,
-                      example=['0', '1', '2'],
-                      default_value='1')
+    parser = argparse.ArgumentParser(
+        description=(
+            "Motion correction of dMRI data. Some of the features to improve robustness were proposed in Xu et al. "
+            "(http://dx.doi.org/10.1016/j.neuroimage.2012.11.014) and include:\n"
+            "  - group-wise (-g)\n"
+            "  - slice-wise regularized along z using polynomial function (-param). For more info about the method, "
+            "type: isct_antsSliceRegularizedRegistration\n"
+            "  - masking (-m)\n"
+            "  - iterative averaging of target volume\n"
+        ),
+        formatter_class=SmartFormatter,
+        add_help=None,
+        prog=os.path.basename(__file__).strip(".py")
```

Comment:
Is this snippet required? In e.g. `sct_run_batch` the `prog` argument isn't set explicitly and the usage comes out correct.

Reply:
you're right-- maybe that's a collateral benefit of having added the scripts as entry point in the setup.py (in the past, that snippet was required)

Reply:
As far as I understand, setting `prog` here is done to avoid the extra `.py` in the help message, right? Unsetting it shows `sct_dmri_moco.py` instead of `sct_dmri_moco`.

> Is this snippet required? In e.g. sct_run_batch the prog argument isn't set explicitly and the usage comes out correct.

Hmm, where are you seeing it not being set explicitly?

https://github.com/neuropoly/spinalcordtoolbox/blob/bf3de687c20ac86717de3584296af1931e247784/scripts/sct_run_batch.py#L55